keyboards/ergotravel/keymaps/default/keymap.c

```diff
@@ -0,0 +1,141 @@
+      GRAVE, 1  , 2  , 3  , 4  , 5  , A,           B,  6  , 7  , 8  , 9  , 0  ,DEL,
+    //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+       TAB , F1 , F2 , F3 , F4 , F5 , F6 ,         D,  Y  ,MINS, EQL,LBRC,RBRC,BSLS,
+    //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+       LSFT, F7 , F8 , F9 , F10, F11, F12,       SPC,  N  , M ,COMM,DOT ,SLSH,ENT ,
+    //|----+----+----+----+----+----+----.     .----+----+----+----+----+----+----|
+       LCTL,LGUI,LALT,ADJT,LOWR,SPC,                 SPC, RASE,LEFT, UP ,DOWN,RIGHT
+    //`----+----+----+--+-+----/----/               \----\----+----+----+----+----'
+  ),
+
+  [_RAISE] = KC_KEYMAP(
+    //,----+----+----+----+----+----+----.     ,----+----+----+----+----+----+----.
+       ESC , 1  , 2  , 3  , 4  , 5  , A,          B,  6  , 7  , 8  , 9  , 0  ,BSPC,
+    //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+       TAB , Q  , W  , E  , R  , T  , C ,         D,  Y  , U  , I  , O  , P  ,DEL ,
+    //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+       LSFT, Z  , X  , C  , V  , B  , A,        SPC,  N  , M  ,COMM,DOT ,SLSH,ENT ,
+    //|----+----+----+----+----+----+----.     .----+----+----+----+----+----+----|
+       LCTL,LGUI,LALT,ADJT,LOWR,SPC,                 SPC, RASE,LEFT, UP ,DOWN,RIGHT
+    //`----+----+----+--+-+----/----/               \----\----+----+----+----+----'
+  ),
+
+  [_ADJUST] = KC_KEYMAP(
+    //,----+----+----+----+----+----+----.     ,----+----+----+----+----+----+----.
+       ESC , F1 , F2 , F3 , F4 , F5 , F6,         F7,  F8, F9 ,F10 , F11, F12,BSPC,
+    //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+       TAB , Q  , W  , E  , R  , T  , C ,         D,  Y  , U  , I  , O  , P  ,DEL ,
+    //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+       LSFT, Z  , X  , C  , V  , B  , A,        SPC,  N  , M  ,COMM,DOT ,SLSH,ENT ,
+    //|----+----+----+----+----+----+----.     .----+----+----+----+----+----+----|
+       LCTL,LGUI,LALT,ADJT,LOWR,SPC,                 SPC, RASE,LEFT, UP ,DOWN,RIGHT
+    //`----+----+----+--+-+----/----/               \----\----+----+----+----+----'
+  )
+
+};
+
+   
+
+#ifdef AUDIO_ENABLE
+float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
+#endif
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          PLAY_SONG(tone_qwerty);
+        #endif
+        persistent_default_layer_set(1UL<<_QWERTY);
+      }
+      return false;
+      break;
+    case LOWER:
+      if (record->event.pressed) {
+        layer_on(_LOWER);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
```

Comment:
(OPTIONAL)  If you want, you can remove the custom RAISE/LOWER/ADJUST keycodes, in favor of using: 

```c
uint32_t layer_state_set_user(uint32_t state) {
  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
}
```

This will activate the tri layer code, regardless of how the board got to that layer.  So you can use MO(), TG(), LT(), etc, if you use the above.

Reply:
I am not sure how I would go about this.  I can't simply replace a case statement with a function definition...

Reply:
You'd remove the raise/lower functions, and add the `layer_state_set_user` above the `process_record_user` function.   

https://github.com/qmk/qmk_firmware/blob/master/keyboards/planck/keymaps/default/keymap.c#L181-L185

And you'd want to use actual layer codes for the RAISE/LOWER (such as "MO"). 

https://github.com/qmk/qmk_firmware/blob/master/keyboards/planck/keymaps/default/keymap.c#L41

The caveat to  this method is that the ADJUST layer is then only accessible by using this method.